### PR TITLE
Copilot: Add Vim to docs to make it clear it supports both Vim/Neovim

### LINKED ISSUE #24624: Update Copilot Neovim docs to include Vim as well
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/copilot/overview-of-github-copilot/about-github-copilot-for-individuals
https://docs.github.com/en/copilot/getting-started-with-github-copilot/getting-started-with-github-copilot-in-neovim
https://docs.github.com/en/copilot/configuring-github-copilot/configuring-github-copilot-in-neovim
https://docs.github.com/en/copilot/troubleshooting-github-copilot/troubleshooting-github-copilot-in-neovim

### What part(s) of the article would you like to see updated?

Add Vim-specific information alongside Neovim ones. Rename most other references of "Neovim" to "Vim/Neovim".

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by ychin**
### Why:

The repository for Copilot (Neo)Vim plugin was updated to support Vim a while ago. Update the docs to reflect that.

Repository link: https://github.com/github/copilot.vim

Closes: #24624 

### What's being changed (if available, include any code snippets, screenshots, or gifs):

~~I didn't rename the files for now, as it didn't seem important enough.~~ Turned out I need to rename the files to match the titles to pass tests. Ran `script/reconcile-filenames-with-ids.js` to rename the files.

Just changed the title and texts ("Neovim" → "Vim/Neovim"), and added Vim-specific information (mininum version and correct installation procedures pulled from the [repo's page](https://github.com/github/copilot.vim)).

### Check off the following:

- [x] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Comment by AS0820075980** [comment]
Chaiwat oppo A16

ในวันที่ พฤ. 23 มี.ค. 2023 18:41 น. ชัยวัฒน์ จันทร์ทักษ์ <
***@***.***> เขียนว่า:

**Comment by AS0820075980** [comment]
Chaiwat oppo A16

ในวันที่ พฤ. 23 มี.ค. 2023 19:10 น. Yee Cheng Chin ***@***.***>
เขียนว่า:

**Comment by cmwilson21** [comment]
@ychin Thank you for opening a PR and linking it to your issue! ✨ 

I'll get this triaged for review! :zap:

**Comment by ychin** [comment]
Hi! I'm curious if there is a timeline for review? I'm now getting merge conflicts. I can resolve them but just wanted to see if there will be a reviewer scheduled to take a look. Thanks.

**Comment by cmwilson21** [comment]
@ychin Thanks for checking in on this. We are working through our backlog, so we appreciate your patience ❤️ 

Someone will review it soon! 👍

File: lib/all-tools.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ export const allTools = {
   importer_cli: 'GitHub Enterprise Importer CLI',
   javascript: 'JavaScript',
   jetbrains: 'JetBrains IDEs (Beta)',
-  neovim: 'Neovim',
+  vimneovim: 'Vim/Neovim',
```
Comment (hectorsector) [inline lib/all-tools.js:12]: We're good on the addition of Vim/Neovim as a tool for the picker from an Engineering POV. However, I think we'll still need to get a review on the Copilot-specific content changes @cmwilson21

**Review by hectorsector — APPROVED** [review]

File: lib/all-tools.js (hunk 1 above)
Comment (hectorsector) [reply lib/all-tools.js:12]: Thank you for getting this opened @ychin!

**Comment by ychin** [comment]
@cmwilson21 Per the comment above, I think I still need someone from Copilot to take a look per @hectorsector 's comment?

File: lib/all-tools.js (hunk 1 above)
Comment (ychin) [reply lib/all-tools.js:12]: I'm not really sure who's responsibility it is to do that? Should I ping someone to do so or someone on your team will do so?

**Comment by cmwilson21** [comment]
@ychin Thanks for checking in! This PR is on the board and waiting for review. We appreciate your patience as we work through some of our backlog. Someone will have eyes on it soon! 💖

**Review by jules-p — APPROVED** [review]
Hey @ychin :wave: thank you for opening this PR, it looks great :sparkles: I'll get these changes merged for you.

**Pull request merged**